skada/deep/base.py

```diff
@@ -923,3 +889,507 @@ def get_loss(self, y_pred, y_true, X, *args, **kwargs):
+            X = torch.cat((self.X, other.X))
+            y = torch.cat((self.y, other.y))
+            sample_domain = torch.cat((self.sample_domain, other.sample_domain))
+            if keep_weights:
+                sample_weight = torch.cat((self.sample_weight, other.sample_weight))
+            else:
+                sample_weight = _EMPTY_
+            if out:
+                return DeepDADataset(X, y, sample_domain, sample_weight, str(X.device))
+            else:
+                self.X = X
+                self.y = y
+                self.sample_domain = sample_domain
+                self.sample_weight = sample_weight
+                self.has_y = torch.cat((self.has_y, other.has_y))
+                self.has_weights = bool(len(self.sample_weight))
+                self.device = str(X.device)
+                assert self._is_correct(), (
+            "Every input data X should have a domain associated, a label (if there is "
+            f"no label, it must be represented by {_NO_LABEL_}) and, "
+            "optionally, a weight."
+        )
+        else:
+            raise TypeError("Can only merge two instances of DeepDADataset")
+
+    def is_empty(self):
+        return not bool(len(self))
+
+    def _is_correct(self):
+        """Validates the dataset. Fails if there are not as many labels, domain,
+        or weights (if any) as data samples.
+        """
+        return (len(self.X) == len(self.y) == len(self.sample_domain)) and (
+            len(self.X) == len(self.sample_weight) or not self.has_weights) and (
+            self.y.ndim == self.sample_domain.ndim == 1) and (
+            not self.has_weights or self.sample_weight.ndim == 1)
+
+    def __len__(self):
+        assert self._is_correct(), (
+            "Every input data X should have a domain associated, a label "
+            f"(if there is no label, it must be represented by {_NO_LABEL_}) and,"
+            " optionally, a weight."
+        )
+        return len(self.X)
+
+    def __getitem__(self, index):
+        try:
+            X = {"X": self.X[index], "sample_domain": self.sample_domain[index]}
+            
+            if self.has_weights:
+                X["sample_weight"] = self.sample_weight[index]
+            
+            return X, self.y[index]
+        except IndexError:            
+            raise IndexError(
+                "DeepDADataset indices must be supported by Pytorch indexing methods"
+            )
+
+    def add_domain(self, other):
+        """Adds a domain to the dataset.
+
+        Args:
+            other (DeepDADataset, dict, list, tuple(torch.Tensor, np.array)):
+                The domain to add. Should be a data type convertible to DeepDADataset.
+        
+        Raises:
+            ValueError: if the other is not convertible to DeepDADataset.
+
+        Returns:
+            DeepDADataset: the original dataset with the new domain added.
+        """
+        if isinstance(other, dict):
+            try:
```

Comment:
you are testing if other is a dict so it should work no ? And if it's not it should return a error in DeepDADataset no ?